pkg/agent/util/net_windows.go

```diff
@@ -70,19 +85,87 @@ type Route struct {
 	RouteMetric       int
 }
 
-func (r Route) String() string {
+func (r *Route) String() string {
 	return fmt.Sprintf("LinkIndex: %d, DestinationSubnet: %s, GatewayAddress: %s, RouteMetric: %d",
 		r.LinkIndex, r.DestinationSubnet, r.GatewayAddress, r.RouteMetric)
 }
 
-func (r Route) Equal(x Route) bool {
+func (r *Route) Equal(x Route) bool {
 	return x.LinkIndex == r.LinkIndex &&
 		x.DestinationSubnet != nil &&
 		r.DestinationSubnet != nil &&
-		x.DestinationSubnet.IP.Equal(r.DestinationSubnet.IP) &&
+		x.DestinationSubnet.String() == r.DestinationSubnet.String() &&
 		x.GatewayAddress.Equal(r.GatewayAddress)
 }
 
+func (r Route) toMibIpForwardRow() *antreasyscall.MibIpForwardRow {
+	row := antreasyscall.InitializeIpForwardRow()
+	row.DestinationPrefix = *antreasyscall.NewAddressPrefixFromIPNet(r.DestinationSubnet)
+	row.NextHop = *antreasyscall.NewRawSockAddrInetFromIP(r.GatewayAddress)
+	row.Metric = uint32(r.RouteMetric)
+	row.Index = uint32(r.LinkIndex)
+	return row
+}
+
+func (r *Route) Add() error {
+	row := r.toMibIpForwardRow()
+	if err := createIpForwardEntry(row); err != nil {
+		return fmt.Errorf("failed to create new IPForward row: %v", err)
+	}
+	return nil
+}
```

Comment:
Why making them methods of `Route`, instead of just putting the code under `NewNetRoute`. I feel the current way makes it not very straighforward and unnecessaily breaks consistency and couple the data struct with external dependencies.

Reply:
Done, put to `NewNetRoute` similar to when using cmd.